code/obj/machinery/singularity/singularity.dm

```diff
@@ -29,67 +29,61 @@ proc/singularity_containment_check(turf/center)
-
 	plane = PLANE_DEFAULT_NOWARP
 	icon = 'icons/effects/64x64.dmi'
 	icon_state = "whole"
 	anchored = ANCHORED
-	density = 1
+	density = TRUE
 	event_handler_flags = IMMUNE_SINGULARITY | IMMUNE_TRENCH_WARP
 	deconstruct_flags = DECON_NONE
 	flags = 0 // no fluid submerge images and we also don't need tgui interactability
-
-
 	pixel_x = -16
 	pixel_y = -16
-
-	var/has_moved
-	var/active = 0 //determines if the singularity is contained
-	var/energy = 10
-	var/Dtime = null
-	var/Wtime = 0
-	var/dieot = 0
-	var/selfmove = 1
-	var/grav_pull = 6
-	var/radius = 0 //the variable used for all calculations involving size.this is the current size
-	var/maxradius = INFINITY//the maximum size the singularity can grow to
-	var/restricted_z_allowed = FALSE
-	var/right_spinning //! boolean for the spaghettification animation spin direction
-	///Count for rate-limiting the spaghettification effect
-	var/spaget_count = 0
+	var/contained = FALSE //! Value which stores whether it is currently under containment
+	var/stored_energy = 10 //! Amount of energy the singularity has absorbed from objects. If at or below 0, the singularity dissapears
+	var/duration_to_live = null //! The amount of time in seconds the singularity will live for before expiring, in seconds. If null, it lives forever.
+	var/worldtime_spawned = null //! The world.time that this singularity was spawned. Used to keep track of time to live for event singularities
+	var/die_over_time = FALSE //! If enabled, the singularity will slowly lose stored_energy and be deleted when it runs at or below 0.
+	var/self_move = TRUE //! Controls whether or not the singularity moves on its own.
+	var/gravity_pull_radius = 6 //! The initial radius, in tiles, that the singularity will pull objects within. Changes with radius
+	var/radius = 0 //! the variable used for all calculations involving radius.
+	var/max_radius = INFINITY //! the maximum radius the singularity can grow to. Overriden on init to a sensible value
+	var/restricted_z_allowed = FALSE //! Whether to annihilate this singularity if it is spawned on restricted z-levels. You can probably guess why
+	var/right_spinning = null //! boolean for the spaghettification animation spin direction
+	var/spaget_count = 0 //! Count for rate-limiting the spaghettification effect
 	var/katamari_mode = FALSE //! If true the sucked-in objects will get stuck to the singularity
 	var/num_absorbed = 0 //! Number of objects absorbed by the singularity
 	var/num_absorbed_players = 0 //! number of players absorbed
-	var/gib_mobs = 0 //! if it should call gib on mobs
-	var/list/obj/succ_cache
-
-	/// Targeted turf when loose
-	var/turf/target_turf
-	/// How many steps we'll continue to walk towards the target turf before rerolling
-	var/target_turf_counter = 0
+	var/gib_mobs = FALSE //! if it should call gib on mobs
+	var/list/datum/succ_cache //! A list of object types consumed by the singularity, used to remember what it ate for diminishing returns on energy.
```

Comment:
```suggestion
	var/list/succ_cache //! An associative list of object types consumed to the number of times eaten, used for diminishing returns on energy.
```
`/datum` is an instance type, we don't have a `type` type in DM so we just have to rely on comments describing the shape and type of the assoc list.